src/test/java/lambda/part2/exercise/Exercise1.java

```diff
@@ -31,33 +31,42 @@ public void sameAgesCheckerUsingBiPredicate() {
 
         // TODO создать переменную sameAgesChecker: (Person, Person) -> boolean, используя BiPredicate
 
-        // assertTrue(sameAgesChecker.test(person1, person2));
-        // assertFalse(sameAgesChecker.test(person1, person3));
-        // assertFalse(sameAgesChecker.test(person2, person3));
+        BiPredicate<Person, Person> sameAgesChecker =
+                (firstPerson, secondPerson) -> Integer.compare(firstPerson.getAge(), secondPerson.getAge()) == 0;
 
-        // FIXME удалить при реализации
-        throw new UnsupportedOperationException("Not implemented");
+         assertTrue(sameAgesChecker.test(person1, person2));
+         assertFalse(sameAgesChecker.test(person1, person3));
+         assertFalse(sameAgesChecker.test(person2, person3));
     }
 
     // TODO метод getFullName: Person -> String, извлекающий из объекта Person строку в формате "имя фамилия".
-    // private static ... getFullName(...) {
+     private static String getFullName(Person person) {
+        return person.getFirstName() + person.getLastName();
+     }
 
     // TODO метод createExtractorAgeOfPersonWithTheLongestFullName: (Person -> String) -> ((Person, Person) -> int),
     // TODO - принимающий способ извлечения полного имени из объекта Person
     // TODO - возвращающий BiFunction, сравнивающий два объекта Person и возвращающий возраст того, чье полное имя длиннее.
-    // private static ... createExtractorAgeOfPersonWithTheLongestFullName(...) {
+     private static BiFunction<Person, Person, Integer>
+     createExtractorAgeOfPersonWithTheLongestFullName(Function<Person, String> extractor) {
+         return (person1, person2) -> {
+             int person1Length = extractor.apply(person1).length();
+             int person2Length = extractor.apply(person2).length();
+
+             if (person1Length >= person2Length) return person1.getAge();
+             else return person2.getAge();
+         };
+     }
```

Comment:
Can be shorter without separate int variables + with ternary operator.

Reply:
I changed if/else to ternary operator. 
But after reading Clean Code I think that using variables is more readable in this case.